packages/create-instance/create-component-stubs.js

```diff
@@ -60,10 +60,25 @@ function getCoreProperties(componentOptions: Component): Object {
 }
 
 function createClassString(staticClass, dynamicClass) {
-  if (staticClass && dynamicClass) {
-    return staticClass + ' ' + dynamicClass
+  // :class="someComputedObject" can return a string, object or undefined
+  // if it is a string, we don't need to do anything special.
+  let evaluatedDynamicClass = dynamicClass
+
+  // if it is an object, eg { 'foo': true }, we need to evaluate it.
+  // see https://github.com/vuejs/vue-test-utils/issues/1474 for more context.
+  if (typeof dynamicClass === 'object') {
+    evaluatedDynamicClass = Object.keys(dynamicClass).reduce((acc, key) => {
+      if (dynamicClass[key] === true) {
+        return acc + ' ' + key
+      }
+      return acc
+    }, '')
+  }
+
+  if (staticClass && evaluatedDynamicClass) {
+    return staticClass + ' ' + evaluatedDynamicClass
   }
-  return staticClass || dynamicClass
+  return staticClass || evaluatedDynamicClass
```

Comment:
Wouldnt have pushing all to an array and using `join(' ')` at the end been a more cleaner way of doing this?

Reply:
I think both are fine - was having problems iterating over `Object.keys`, using `of` throws some warning. A few other places in the codebase do `Object.keys().reduce` so I just tried to follow that same style.

I like `reduce` but I don't mind either way.